### PR TITLE
Cluster API support

### PR DESCRIPTION
**Pull request by ElliottKasoar**
Replaces Magnum with Cluster API, as well as updating the cinder storage and GPU driver images and minor fixes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
-  * [Priority Class for GPU-operator](#priority-class-for-gpu-operator)
 - [Kubectl Namespaces](#kubectl-namespaces)
 - [Jupyter Hub Config](#jupyter-hub-config)
   * [HTTPS Config](#https-config)
     + [Setting up DNS for Lets Encrypt](#setting-up-dns-for-lets-encrypt)
     + [Using existing TLS Certificate](#using-existing-tls-certificate)
-  * [Using Oauth Sign in (`config-oauth.yaml.template`)](#using-oauth-sign-in---config-oauthyamltemplate--)
-    + [Setting users / admin groups for Oauth](#setting-users---admin-groups-for-oauth)
-  * [Using Native Authenticator (`config-native.yaml.template`)](#using-native-authenticator---config-nativeyamltemplate--)
+  * [Using Oauth Sign in (`config-oauth.yaml.template`)](#using-oauth-sign-in-config-oauthyamltemplate)
```
Comment (DavidFair) [inline README.md:15]: (We need to remove this section as it's not really supported of tested, but for a future PR)

**Review by ElliottKasoar — COMMENTED** [review]
Should we delete roles/deploy_jhub/files/nvidia-priority-class.yaml here too?

File: roles/deploy_jhub/tasks/main.yml [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
+    repo: "https://github.com/kubernetes/cloud-provider-openstack.git"
+    dest: "/tmp/cloud-provider-openstack"
+    version: "release-1.24"
+    update: true # Automatically pull bug-fixes in
+
+- name: Install Cinder CSI
+  kubernetes.core.helm:
+    name: cinder-csi
+    namespace: kube-system
+    chart_ref: "/tmp/cloud-provider-openstack/charts/cinder-csi-plugin"
```
Comment (DavidFair) [inline roles/deploy_jhub/tasks/main.yml:12]: We need to add the Openstack Manager steps as we cannot deploy without the LB support it provides:
https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/openstack-cloud-controller-manager/using-openstack-cloud-controller-manager.md

(I think the actual steps are in our Cluster API Docs)

Assume that the user has populated the secret, or we could double check it with Ansible (not sure how though)

**Comment by ElliottKasoar** [comment]
I can't approve @DavidFair's commits as I created the PR, but reviewed and lgtm.

File: roles/deploy_jhub/tasks/main.yml [hunk 3]
```diff
@@ -11,6 +11,16 @@
     namespace: kube-system
     chart_ref: "/tmp/cloud-provider-openstack/charts/cinder-csi-plugin"
 
+- name: Install Openstack Controller Manager
```
Comment (DavidFair) [inline roles/deploy_jhub/tasks/main.yml:14]: Do we `wget` these into `tmp` before running this?

Comment (ElliottKasoar) [reply roles/deploy_jhub/tasks/main.yml:14]: It's in the same repo as the cinder installation just above, so we use that

**Review by DavidFair — APPROVED** [review]
LGTM, thanks for tidying up the last little bits :+1:

**Pull request merged**